### PR TITLE
System by name

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Add alias URL to show system details by name like  `https://mysrv/rhn/manager/systems/details/by-name/<name>`. Will redirect to the SID-based page.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes
- [X] **DONE**

## Test coverage
- No tests: to be used by cucumber tests to speed them up

- [X] **DONE**

## Links

Fixes #
Tracks # **add downstream PR, if any**

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by rjmateus — APPROVED** [review]
LGTM

**Review by mcalmer — APPROVED** [review]

**Comment by cbosdo** [comment]
@mcalmer @rjmateus I changed the way to get the page using the system name to avoid redirect. Sadly there are a big number of places where we use the SID in ACLs... This is a big sub-optimized mess!
Which variant to you prefer?

**Comment by rjmateus** [comment]
I'm not sure about this one. We are adding complexity to the product for something the user will not need. This will be used only for the test suite, which will be using a code path different from the one used by the user.

The goal is to reduce the need to access so many pages in the test suite to be able to reach the system details page. So I would say shortcuts should be taken on the test suite side to speed up the test.

Some options to improve on the test suite side:
- Cache the system ID. In the first access to the normal navigation and save the system ID in an internal cache variable. Next access reads the ID from the cache and makes direct access to the page.
- Get the ID from the database. Run a remote command on the server (some commands are already being run there) that executes a DB query to extract the system id.

Any thoughts on this?

File: java/code/src/com/redhat/rhn/frontend/action/systems/sdc/SystemOverviewAction.java [hunk 1]
```diff
@@ -65,8 +67,19 @@ public class SystemOverviewAction extends RhnAction {
...
         RequestContext rctx = new RequestContext(request);
-        Long sid = rctx.getRequiredParam("sid");
         User user = rctx.getCurrentUser();
+        Long sid = null;
+        String name = rctx.getParam("name", false);
+        if (name != null) {
+            List<SystemOverview> matches = SystemManager.listSystemsByName(user, name);
+            if (matches.size() != 1) {
+                throw new BadParameterException("No or multiple systems matching the name");
+            }
```
Comment (srbarrios) [inline java/code/src/com/redhat/rhn/frontend/action/systems/sdc/SystemOverviewAction.java:77]: image
Good to catch this :)

**Comment by cbosdo** [comment]
It has been decided to not add code only for testing purpose. The testsuite can just get the SID using API and call the URL as a first step and possibly cache it if needed.

**Pull request closed without merging**